cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -288,41 +280,44 @@ def file_exists_in_latest_version_for_bundle(
             file_path.name == Path(bundle_file.path).name for bundle_file in latest_version.files
         )
 
-    def create_flow_cell(self, parsed_flow_cell: FlowCellDirectoryData) -> Flowcell:
+    def store_flow_cell_in_status_db(self, parsed_flow_cell: FlowCellDirectoryData) -> None:
         """Create flow cell from the parsed and validated flow cell data."""
-        return Flowcell(
+        flow_cell: Flowcell = Flowcell(
             name=parsed_flow_cell.id,
             sequencer_type=parsed_flow_cell.sequencer_type,
             sequencer_name=parsed_flow_cell.machine_name,
             sequenced_at=parsed_flow_cell.run_date,
         )
+        self.status_db.session.add(flow_cell)
+        self.status_db.session.commit()
 
-    def parse_and_validate_flow_cell_directory_data(
+    def parse_flow_cell_directory_data(
         self, flow_cell_directory: Path, bcl_converter: str
     ) -> FlowCellDirectoryData:
         """Parse flow cell data from the flow cell directory."""
+        if not self.is_flow_cell_directory_valid(flow_cell_directory=flow_cell_directory):
+            raise FlowCellError(f"Flow cell directory not found: {flow_cell_directory}")
+
         try:
             flow_cell: FlowCellDirectoryData = FlowCellDirectoryData(
                 flow_cell_path=flow_cell_directory, bcl_converter=bcl_converter
             )
             return flow_cell
-
-        except FlowCellError:
+        except FlowCellError as e:
```

Comment:
```suggestion
        except FlowCellError as error:
```
or exception? e is so ambigous